Update CHANGELOG.md for #1736 (#1750)
hashicorp/boundary: CHANGELOG.md

* config: Add support for go-sockaddr templates to Worker and Controller
  addresses. ([PR](https://github.com/hashicorp/boundary/pull/1731))
* host: Plugin-based host catalogs will now schedule updates for all

>   host sets contained within its attributes are updated.

Reviewer: ```suggestion
  of its host sets when its attributes are updated.
```
